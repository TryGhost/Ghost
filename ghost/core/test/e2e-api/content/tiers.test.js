const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');

describe('Tiers Content API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getContentAPIAgent();
        await fixtureManager.init('members', 'api_keys', 'tiers:archived', 'tiers:hidden');
        await agent.authenticate();
    });

    it('Can request only active tiers', async function () {
        await agent.get('/tiers/?include=monthly_price')
            .expectStatus(200)
            .matchHeaderSnapshot({
                'content-version': matchers.anyContentVersion,
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                tiers: Array(3).fill({
                    id: matchers.anyObjectId,
                    created_at: matchers.anyISODate,
                    updated_at: matchers.anyISODate
                })
            });
    });

    it('Can filter on visibility', async function () {
        await agent.get('/tiers/?filter=visibility:public')
            .expectStatus(200)
            .matchHeaderSnapshot({
                'content-version': matchers.anyContentVersion,
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                tiers: Array(2).fill({
                    id: matchers.anyObjectId,
                    created_at: matchers.anyISODate,
                    updated_at: matchers.anyISODate
                })
            });
    });
});
