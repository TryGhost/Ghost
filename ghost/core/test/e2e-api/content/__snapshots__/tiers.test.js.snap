// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiers Content API Can filter on visibility 1: [body] 1`] = `
Object {
  "meta": Object {
    "pagination": Object {
      "limit": 2,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 2,
    },
  },
  "tiers": Array [
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Free",
      "slug": "free",
      "trial_days": 0,
      "type": "free",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-free",
    },
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "currency": "USD",
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "monthly_price": 500,
      "name": "Default Product",
      "slug": "default-product",
      "trial_days": 0,
      "type": "paid",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-paid",
      "yearly_price": 5000,
    },
  ],
}
`;

exports[`Tiers Content API Can filter on visibility 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "public, max-age=0",
  "content-length": "727",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Tiers Content API Can request only active tiers 1: [body] 1`] = `
Object {
  "meta": Object {
    "pagination": Object {
      "limit": 3,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 3,
    },
  },
  "tiers": Array [
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Free",
      "slug": "free",
      "trial_days": 0,
      "type": "free",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-free",
    },
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "currency": "USD",
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "monthly_price": 500,
      "name": "Default Product",
      "slug": "default-product",
      "trial_days": 0,
      "type": "paid",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-paid",
      "yearly_price": 5000,
    },
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "currency": "USD",
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "monthly_price": 500,
      "name": "product",
      "slug": "gold-2",
      "trial_days": 0,
      "type": "paid",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "none",
      "welcome_page_url": null,
      "yearly_price": 5000,
    },
  ],
}
`;

exports[`Tiers Content API Can request only active tiers 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "public, max-age=0",
  "content-length": "1051",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
